ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtil.java

```diff
@@ -157,85 +157,92 @@ public static double calc_total_balance(BeaconState state) {
-      long current_committees_per_epoch = get_current_epoch_committee_count(state);
+    } else if (epoch.compareTo(next_epoch) == 0) {
+      UnsignedLong current_committees_per_epoch = get_current_epoch_committee_count(state);
       committees_per_epoch = get_next_epoch_committee_count(state);
       shuffling_epoch = next_epoch;
-      long epochs_since_last_registry_update =
-          current_epoch - state.getValidator_registry_update_epoch().longValue();
-      // TODO: convert these values to UnsignedLong
+      UnsignedLong epochs_since_last_registry_update =
+          current_epoch.minus(state.getValidator_registry_update_epoch());
       if (registry_change) {
         seed = generate_seed(state, next_epoch);
         shuffling_start_shard =
-            (state.getCurrent_epoch_start_shard().longValue() + current_committees_per_epoch)
-                % Constants.SHARD_COUNT;
-      } else if (epochs_since_last_registry_update > 1
+            state
+                .getCurrent_epoch_start_shard()
+                .plus(current_committees_per_epoch)
+                .mod(UnsignedLong.valueOf(Constants.SHARD_COUNT));
+      } else if (epochs_since_last_registry_update.compareTo(UnsignedLong.ONE) > 0
           && is_power_of_two(epochs_since_last_registry_update)) {
         seed = generate_seed(state, next_epoch);
-        shuffling_start_shard = state.getCurrent_epoch_start_shard().longValue();
+        shuffling_start_shard = state.getCurrent_epoch_start_shard();
       } else {
         seed = state.getCurrent_epoch_seed();
-        shuffling_start_shard = state.getCurrent_epoch_start_shard().longValue();
+        shuffling_start_shard = state.getCurrent_epoch_start_shard();
       }
     }
 
+    // TODO: revist when we have the final shuffling algorithm implemented
     List<List<Integer>> shuffling =
-        get_shuffling(seed, state.getValidator_registry(), shuffling_epoch);
-    long offset = slot % Constants.EPOCH_LENGTH;
-    long committees_per_slot = committees_per_epoch / Constants.EPOCH_LENGTH;
-    long slot_start_shard =
-        (shuffling_start_shard + committees_per_slot * offset) % Constants.SHARD_COUNT;
+        get_shuffling(seed, state.getValidator_registry(), shuffling_epoch.longValue());
+    UnsignedLong offset = slot.mod(UnsignedLong.valueOf(Constants.EPOCH_LENGTH));
+    UnsignedLong committees_per_slot =
+        committees_per_epoch.dividedBy(UnsignedLong.valueOf(Constants.EPOCH_LENGTH));
+    UnsignedLong slot_start_shard =
+        shuffling_start_shard
+            .plus(committees_per_slot)
+            .times(offset)
+            .mod(UnsignedLong.valueOf(Constants.SHARD_COUNT));
 
     ArrayList<CrosslinkCommittee> crosslink_committees_at_slot =
         new ArrayList<CrosslinkCommittee>();
-    for (int i = 0; i < committees_per_slot; i++) {
+    // TODO: revist when we have the final shuffling algorithm implemented
+    for (long i = 0; i < committees_per_slot.longValue(); i++) {
```

Comment:
There's the potential for this to 2s complement overflow into a negative number, causing the loop to never run. Perhaps a bounds check here? Also we probably need a more comprehensive approach to iterating with `UnsignedLong`s as indices rather than just using `UnsignedLong#longValue`.

Reply:
agreed...that's why i had to put a comment there.  this all has to be rewritten and since it will be touched again when the shuffling algorithm is implemented, then it can be handled then.  right now there is no point in messing with it